aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp

```diff
@@ -17,17 +17,50 @@ namespace {
 class QConvPackWeightInt8 final : public c10::OperatorKernel {
  public:
 #ifdef USE_FBGEMM
-  Tensor operator()(Tensor weight, int64_t groups) {
+  Tensor operator()(
+      Tensor weight,
+      torch::List<int64_t> stride,
+      torch::List<int64_t> padding,
+      torch::List<int64_t> dilation,
+      int64_t groups) {
     TORCH_CHECK(
         weight.ndimension() == 4, "Weights are expected to have 4 dimensions");
+
+    TORCH_CHECK(stride.size() == 2, "2D convolution only");
+    TORCH_CHECK(
+        padding.size() == 2,
+        "Specify top/left padding only. \
+        bottom/right padding assumed to be equal to top/left");
+    TORCH_CHECK(dilation.size() == 2, "2D convolution only");
+    TORCH_CHECK(
+        (dilation[0] == 1 && dilation[1] == 1),
+        "Currently dilation should be 1");
     // weights in KRS(C/G) format
-    // matrix dimensions after im2col
     int output_channels = weight.size(0);
     int kernel_h = weight.size(1);
     int kernel_w = weight.size(2);
     int input_channels_per_group = weight.size(3);
-    int NDim = output_channels / groups;
-    int KDim_per_group = kernel_h * kernel_w * input_channels_per_group;
+
+    // mini-batch doesn't have any impact on how we pack weights
+    // so we pass it as 1
+    // Input image height/width also don't have any impact on how we pack
+    // weights so we can pass any values
+    fbgemm::conv_param_t<2> conv_p(
+        1, // Mini-Batch
+        input_channels_per_group * groups, // input channels
+        output_channels,
```

Comment:
Why do we need mini-batch and image size as parameters to this function if they do not have any effect? Can we remove these arguments?

Reply:
The interface for packing takes struct conv_param_t as argument. This way we pass the same conv_param_t struct to both conv operation as well as packing. We will have to change the interface and define a new struct if we want to remove these parameters and that would just work for packing only. The other way of doing this is to add a function in fbgemm that would return conv_param_t and does't take mini-batch and imagesize as input parameters.